woocommerce/woocommerce-gateway-stripe
# Get correct value of required checkbox from the checkout fields when paying with Google/Apple Pay

Reported in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2886#issuecomment-2093996181 that the value of a required checkbox is not retrieved correctly in the `getRequiredFieldDataFromCheckoutForm` function. This results in always sending `1` as the checkbox value.

## Changes 
- We are now checking if the field is a checkbox before retrieving the value. If it is a checkbox, we get the value from `checked` property of the element.
- This change is done for the shortcode checkout only. For block checkout adding a custom field and relevant validation needs custom code and the custom required field handling could be done within that implementation.

## Testing instructions

1. If you already don't have, then create a shortcode checkout page.
2. Go to **WooCommerce -> Settings -> Payments -> Stripe** and enable Google/Apple Pay on the checkout page.
3. Install the [Checkout Field Editor plugin](https://woocommerce.com/products/woocommerce-checkout-field-editor/) 
4. Go to **WooCommerce > Checkout fields** and create a required checkbox under the `Addtional Fields` tab. Also make the company field required from the `Billing Fields` tab.

![checkbox](https://github.com/woocommerce/woocommerce-gateway-stripe/assets/33387139/ca85bb82-9185-4b8c-9883-ea6a4f3a3ca7)

5. As a shopper, add a product to your cart. Now proceed to the checkout page. Here the Google/Apple Pay button should be present.
6. Try to pay with Google/Apple Pay without filling up the company field and checking the checkbox. You should see an error message on the page about the required fields.
7.  Add any value to the company field and check the checkbox. Try again to pay with Google/Apple Pay. This time the purchase should be successful.
8. Check the order data in your wp-admin and make sure the company name is there. Scroll down to check the custom fields section and make sure the checkbox value is correctly saved.
